camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java

```diff
@@ -163,53 +168,96 @@ public boolean fetch() throws IOException {
     try {
       fetchResponse = simpleConsumer.fetch(fetchRequest);
       if (fetchResponse.hasError()) {
+        throw new RuntimeException();
```

Comment:
This exception is solely for consumption in line 174 (so that we don't need to write duplicate code for fetchResponse=null, fetchResponse.hasError(), and fetch() has exception. All three cases will go to line 174). The error message is printed in line 175.